### DISCUSSION TITLE
Set a default value at inclusion

### DISCUSSION THREAD

**Discussion by sl8vz** [zwave-js/zwave-js#6235] [Q&A]
Hi,
I would like to know if there is way to set a default value for configuration parameter in the model that would be applied when the device is included.  
Indeed, it sometimes happens that the factory configuration is not ideal and being able to enforce a specific value in the model could prevent to do it manually on each device.  
We tried to play with the `defaultValue` but it seems it's only there to provide the factory value. Changing its value doesn't write it to the device at inclusion. Openzwave used to do that btw.  

Thanks,
Sebastien

**Comment by AlCalzone** [comment]
We generally don't reconfigure users' devices (although in some cases it would be better). We could consider defining recommended values for config parameters in config files and adding a setting for users to opt in. Optionally, we could add this to https://github.com/zwave-js/node-zwave-js/issues/2909 so users get notified of values that are outside the recommended range.

For a more short-term solution, Z-Wave JS UI has a feature in the advanced section called "driver function". It ships with a snippet for copying config parameters between devices automatically. Maybe that is good enough for now.
